include/posix/pthread.h

```diff
@@ -7,39 +7,42 @@
 #ifndef __PTHREAD_H__
 #define __PTHREAD_H__
 
-#ifdef CONFIG_NEWLIB_LIBC
+#include <kernel.h>
 #include <time.h>
-#else
-/* This should probably live somewhere else but Zephyr doesn't
- * currently have a stdc layer to provide it
+#include "sys/types.h"
+#include "posix_sched.h"
+#include "unistd.h"
+
+struct posix_thread {
+	struct k_thread thread;
+	void *retval;
+};
+
+/* Pthread cancellation */
+#define PTHREAD_CANCEL_POS     0
+/* Non Essential Thread */
+#define PTHREAD_CANCELABLE     (0 << PTHREAD_CANCEL_POS)
+/* Essential Thread */
+#define PTHREAD_NON_CANCELABLE (1 << PTHREAD_CANCEL_POS)
+#define THREAD_INIT_FLAGS       PTHREAD_CANCELABLE
+
+/*
+ *  Default pthread attributes on attributes initialization.
  */
-struct timespec {
-	s32_t tv_sec;
-	s32_t tv_nsec;
+static const pthread_attr_t initpthreadattrs = {
```

Comment:
Nitpick, but please use `_` to separate words in variable names: `init_pthread_attrs`.

Also, this shouldn't be in the header file; there will be multiple copies of it if it's included multiple times.  Define it inside `pthread_attr_init()` instead, even if you have to move it to a `.c` file.